src/lib.rs

```diff
@@ -89,10 +89,17 @@ pub fn parse_length(buf: &[u8]) -> Result<(LengthByteSize, ContentByteSize), Str
         LengthOctet::Long(x) => match x {
             1 => Ok((2, buf[1] as usize)),
             2 => Ok((3, BigEndian::read_u16(&buf[1..3]) as usize)),
-            4 => Ok((4, BigEndian::read_u32(&buf[1..5]) as usize)),
-            8 => Ok((4, BigEndian::read_u64(&buf[1..9]) as usize)),
+            3 => {
+                // parse uint24 by padding with leading zero
+                let mut buf_tmp = [0_u8; 4];
+                let arr_ref = &mut buf_tmp[1..4];
+                arr_ref.copy_from_slice(&buf[1..4]);
+                Ok((4, BigEndian::read_u32(&buf_tmp) as usize))
+            }
+            4 => Ok((5, BigEndian::read_u32(&buf[1..5]) as usize)),
+            8 => Ok((9, BigEndian::read_u64(&buf[1..9]) as usize)),
```

Comment:
from my understanding this 9 should be the amount of bytes that the length takes so that they can be skipped when parsing the value. So first byte defines that it's the long BER encoding and the following 8 correspond to the length and so we have to skip 9 bytes.